threedi_modelchecker/tests/test_checks_other.py

```diff
@@ -293,6 +294,69 @@ def test_cross_section_location(session):
     assert len(errors) == 1
 
 
+@pytest.mark.parametrize(
+    "shape, width, height, ok",
+    [
+        # --- closed cross-sections ---
+        # shapes 0, 2, 3 and 8 are always closed
+        *[
+            (
+                i,
+                "0 4.142 5.143 5.143 5.869 5.869",
+                "0 0.174 0.348 0.522 0.696 0.87",
+                False,
+            )
+            for i in [0, 2, 3, 8]
+        ],
+        # shapes 5 and 6 are closed if the width at the highest increment (last number in the width string) is 0
+        *[
+            (i, "0 4.142 5.143 5.143 5.869 0", "0 0.174 0.348 0.522 0.696 0.87", False)
+            for i in [5, 6]
+        ],
+        # shape 7 is closed if the first and last (width, height) coordinates are the same
+        (7, "2 4.142 5.143 5.143 5.869 2", "3 0.174 0.348 0.522 0.696 3", False),
+        #
+        # --- open cross-sections ---
+        # shape 1 is always open
+        (1, "0 4.142 5.143 5.143 5.869 5.869", "0 0.174 0.348 0.522 0.696 0.87", True),
+        # shapes 5 and 6 are open if the width at the highest increment (last number in the width string) is > 0
+        *[
+            (i, "0 4.142 5.143 5.143 5.869 1", "0 0.174 0.348 0.522 0.696 0.87", True)
+            for i in [5, 6]
+        ],
+        # shape 7 is open if the first and last (width, height) coordinates are not the same
+        (7, "2 4.142 5.143 5.143 5.869 2", "3 0.174 0.348 0.522 0.696 4", True),
+    ],
```

Comment:
Can you add a test with wrong strings? Like "foo". There shouldn't be an exception in either case.